src/main/java/uk/gov/companieshouse/company/links/processor/ResponseHandler.java

```diff
@@ -21,18 +21,19 @@ public static void handleResponse(
             final String msg,
             final Map<String, Object> logMap,
             final Logger logger)
-            throws NonRetryErrorException, RetryErrorException {
+            throws NonRetryableErrorException, RetryableErrorException {
         logMap.put("status", httpStatus.toString());
         if (HttpStatus.BAD_REQUEST == httpStatus) {
             // 400 BAD REQUEST status is not retryable
-            logger.errorContext(logContext, msg, null, logMap);
-            throw new NonRetryErrorException(msg);
-        } else if (httpStatus.is4xxClientError() || httpStatus.is5xxServerError()) {
+            throw new NonRetryableErrorException(String
```

Comment:
Final bullet point:
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/16381559/162740956-a5d50a12-5fc7-454a-b79b-048026763385.png">

Reply:
`handleResponse` is only used for the get/patch to company profile, how does this handle the first bullet point about Avro de-/serialisation?

Reply:
I've seen this is done elsewhere :) screenshot gives the wrong impression